src/config.js

```diff
@@ -521,6 +524,75 @@ export default class Config {
     }
   }
 
+  async findManifest(dir: string, isRoot: boolean): Promise<?Manifest> {
+    for (const registry of registryNames) {
+      const manifest = await this.tryManifest(dir, registry, isRoot);
+
+      if (manifest) {
+        return manifest;
+      }
+    }
+
+    return null;
+  }
+
+  async findWorktree(initial: string): Promise<?string> {
+    let previous = null;
```

Comment:
install command is guarded by `projectManifestJson.workspaces && this.config.workspacesExperimental`.
How about extracting it in a setting and checking it here?